dotnet/src/Microsoft.Agents.AI.Abstractions/Features/IAgentFeatureCollection.cs

```diff
@@ -24,23 +25,30 @@ public interface IAgentFeatureCollection : IEnumerable<KeyValuePair<Type, object
     int Revision { get; }
 
     /// <summary>
-    /// Gets or sets a given feature. Setting a null value removes the feature.
+    /// Attempts to retrieve a feature of the specified type.
     /// </summary>
-    /// <param name="key"></param>
-    /// <returns>The requested feature, or null if it is not present.</returns>
-    object? this[Type key] { get; set; }
+    /// <typeparam name="TFeature">The type of the feature to retrieve.</typeparam>
```

Comment:
One downside to removing the indexer is that there is no way to get/remove a feature given a `Type`: we must statically have a `TFeature` to provide. I'm not sure if this will be needed, but it's worth noting.